src/librustc_mir/borrow_check/borrow_set.rs

```diff
@@ -237,6 +237,14 @@ impl<'a, 'gcx, 'tcx> Visitor<'tcx> for GatherBorrows<'a, 'gcx, 'tcx> {
                             TwoPhaseActivation::NotActivated
                         }
                         _ => {
+                            // Double check: We should have found an activation for every pending
```

Comment:
I think this commit is not *quite* correct -- that is, what we are checking here is that (a) this is indeed a 2-phase borrow and (b) we've not found some other activation (though we also check that above), right?

Reply:
Pushed improved comments.